org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/DefaultGradleBuild.java

```diff
@@ -46,7 +49,17 @@ public DefaultGradleBuild(BuildConfiguration buildConfig) {
     @Override
     public void synchronize(NewProjectHandler newProjectHandler, CancellationTokenSource tokenSource, IProgressMonitor monitor) throws CoreException {
         SynchronizeGradleBuildsOperation syncOperation = SynchronizeGradleBuildsOperation.forSingleGradleBuild(this, newProjectHandler);
-        syncOperation.run(tokenSource, monitor);
+        try {
+            syncOperation.run(tokenSource, monitor);
+            GradleMarkerManager.clear(this);
```

Comment:
⭕️ Do we clear markers when unlinking a project?

Reply:
I think we do, but there's no coverage for that. Let me check.

Reply:
I've added coverage for this usecase.